ts/src/profile-serializer.ts

```diff
@@ -344,7 +345,23 @@ export function serializeTimeProfile(
   return new Profile(profile);
 }
 
-function buildLabels(labelSet: LabelSet, stringTable: StringTable): Label[] {
+function getEndPoint(
+  labelSet: LabelSet,
+  rootSpanIdToEndPointMap: Map<string, string>
+) {
+  const localRootSpanId = labelSet['local root span id'];
```

Comment:
On second thought, it does make me sad that we need to veer away from having a generic solution for "labels" and push down knowledge about our specific single usage of this feature into the profiler codebase.

Reply:
Changed the API to be more generic !